Error compiling opencv_dnn

Hi,

I'm trying to compile opencv_dnn and I have got errors : 

```
4>  caffe_converter.cpp
4>F:\Install\protobuf\include\google/protobuf/io/coded_stream.h(864): Warning C4800: 'int' : forcing value to bool 'true' or 'false' (performance warning)
4>F:\Install\protobuf\include\google/protobuf/io/coded_stream.h(868): Warning C4800: 'int' : forcing value to bool 'true' or 'false' (performance warning)
4>g:\lib\opencv\dll2015\3rdparty\tinydnn\tiny-dnn-1.0.0a3\tiny_dnn\io\caffe\caffe.pb.h(2855): error C2059: syntax error: 'constant'
4>g:\lib\opencv\dll2015\3rdparty\tinydnn\tiny-dnn-1.0.0a3\tiny_dnn\io\caffe\caffe.pb.h(2856): error C2238: unexpected token(s) preceding  ';'
4>g:\lib\opencv\dll2015\3rdparty\tinydnn\tiny-dnn-1.0.0a3\tiny_dnn\io\caffe\caffe.pb.h(10019): error C2059:  syntax error: 'constant'
4>g:\lib\opencv\dll2015\3rdparty\tinydnn\tiny-dnn-1.0.0a3\tiny_dnn\io\caffe\caffe.pb.h(10020): error C2238: unexpected token(s) preceding  ';'
4>G:\Lib\opencv\dll2015\3rdparty\tinydnn\tiny-dnn-1.0.0a3\tiny_dnn/io/caffe/layer_factory_impl.h(347): warning C4267: 'argument' : conversion de 'size_t' en 'int', perte possible de données
4>G:\Lib\opencv\dll2015\3rdparty\tinydnn\tiny-dnn-1.0.0a3\tiny_dnn/io/caffe/layer_factory_impl.h(348): warning C4267: 'argument' : conversion de 'size_t' en 'int', perte possible de données
4>G:\Lib\opencv\dll2015\3rdparty\tinydnn\tiny-dnn-1.0.0a3\tiny_dnn/io/caffe/layer_factory_impl.h(392): warning C4267: 'argument' : conversion de 'size_t' en 'int', perte possible de données
4>G:\Lib\opencv\dll2015\3rdparty\tinydnn\tiny-dnn-1.0.0a3\tiny_dnn/io/caffe/layer_factory.h(90): warning C4457: la déclaration de 'layer' masque le paramètre de fonction

```

Error is at this line : 
`  static const DimCheckMode STRICT =
    ParamSpec_DimCheckMode_STRICT;
`
and this one : 
`  static const DimCheckMode STRICT =
    V1LayerParameter_DimCheckMode_STRICT;
`
I installed  protobuf my version is 2017-04-05 version 3.3.0 (C++/Java/Python/PHP/Objective-C/C#/Ruby/JavaScript)

- OpenCV => 3.3.0-rc
- Operating System / Platform => Windows 10 64 Bit
- Compiler => Visual Studio 2015
-->
STRICt is already define in c:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\shared\minwindef.h 

I post same issue on tiny_dnn

--- a/modules/dnn_modern/CMakeLists.txt
+++ b/modules/dnn_modern/CMakeLists.txt
@@ -157,6 +157,7 @@ elseif(MSVC)
     # this is fine
     add_definitions(-D _CRT_SECURE_NO_WARNINGS)
     add_definitions(-D _SCL_SECURE_NO_WARNINGS)
+    add_definitions(-D NO_STRICT)
     # prolly powerless with header-only project
     set(EXTRA_C_FLAGS "${EXTRA_C_FLAGS} /MP")
 endif()

